src/gui/qgsmapcanvas.h

```diff
@@ -555,7 +561,7 @@ class GUI_EXPORT QgsMapCanvas : public QGraphicsView
      * \see setExpressionContextScope()
      * \since QGIS 2.12
      */
-    QgsExpressionContextScope &expressionContextScope() { return mExpressionContextScope; }
+    QgsExpressionContextScope &expressionContextScope();
```

Comment:
Hmmm... a bit puzzled by this method, is this thought to be modified by internal (as you are doing here), or external (returned by reference) and then after all, everything will be overwritten by the setter above.

Maybe we need two scopes, an internal one (which the canvas itself manages and will not "just be overwritten") and an external one that can be modified (in this case, why not via the setter?).

Reply:
@m-kuhn , I don't think I've changed anything in terms of how these functions are served. There value was served as a reference before alongside the setter. 

I don't know if there's a valid reason for that. It's useful here to avoid code duplication (ie appending the @canvas_cursor_point to the scope) between canvas rendering vs external use of canvas' scope (symbology UI for eg).

Reply:
I'm inclined to agree with @m-kuhn here, and the two should be split into an internal private scope + a separate publicly available and modifiable scope (Which is added after the internal scope, so takes precedence)